src/libraries/System.Collections.Immutable/tests/Frozen/FrozenDictionaryAlternateLookupTests.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Xunit;
+
+namespace System.Collections.Frozen.Tests
+{
+    public class FrozenDictionaryAlternateLookupTests
+    {
+        [Fact]
+        public void AlternateLookup_Empty()
+        {
+            Assert.False(FrozenDictionary<string, string>.Empty.TryGetAlternateLookup<ReadOnlySpan<char>>(out _));
```

Comment:
If I'm interpreting this test correctly, the empty singleton does not return an alternate lookup because it's a special implementation however creating an empty frozen dictionary using inputs will return a lookup? I find this surprising, my expectation would have been to get a consistent output for a given set of type parameters.

Reply:
>  the empty singleton does not return an alternate lookup because it's a special implementation however creating an empty frozen dictionary using inputs will return a lookup?

No, it's based on the comparer. The Empty singleton uses `EqualityComparer<T>.Default`, which does not implement the alternate interface. An empty dictionary constructed with `EqualityComparer<T>.Default` would similarly return false here. But when constructing an empty with a comparer that does implement it, e.g. StringComparer.Ordinal, then this returns true, regardless of whether the collection is empty or not.

Reply:
Oh I see, it's because the default comparer for string doesn't support alternate lookup?

Reply:
> Oh I see, it's because the default comparer for string doesn't support alternate lookup?

Right

Reply:
If we address https://github.com/dotnet/runtime/issues/102906, then this will need to be revisited.